### PR TITLE
Rename border agent proxy to TMF proxy

### PR DESCRIPTION
**Pull request by bukepo**
There are new TMF messages except border agent's need to be handled on host side, border agent proxy cannot exactly show the actual functionality.

This PR renames the Border Agent Proxy to TMF Proxy, both in source code and spinel protocol.

### CODE REVIEW

**Comment by jwhui** [comment]
"Border Router Proxy" seems overly generic.  Should we rename to something like "TMF Proxy"?

**Comment by bukepo** [comment]
Agree, I will update to `TMF proxy`.

File: doc/draft-rquattle-spinel-unified.txt [hunk 1]
```diff
@@ -2878,9 +2878,9 @@ Quattlebaum & woodyatt  Expires November 2, 2017               [Page 51]
                | Fields:  | Length | CoAP | locator | port |
                +----------+--------+------+---------+------+
 
-   This property allows the host to send and receive border-agent-
+   This property allows the host to send and receive border-router-
    related CoAP requests/responses from the NCP's RLOC address.  This
-   allows the host driver to implement a Thread border agent.
+   allows the host driver to implement a Thread border router.
```
Comment (jwhui) [inline doc/draft-rquattle-spinel-unified.txt]: how about: "This property allows the host to send and receive TMF messages from the NCP's RLOC address and support Thread-specific border router functions."

File: doc/spinel-protocol-src/spinel-tech-thread.md [hunk 2]
```diff
@@ -286,9 +286,9 @@ Octects: | 2      | *n*  |    2    |  2
 ---------|--------|------|---------|-------
 Fields:  | Length | CoAP | locator | port
 
-This property allows the host to send and receive border-agent-related
+This property allows the host to send and receive border-router-related
```
Comment (jwhui) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: See comment above.

File: examples/Makefile-posix [hunk 3]
```diff
@@ -98,7 +98,7 @@ configure_OPTIONS                   = \
     --with-examples=posix             \
     --with-platform-info=POSIX        \
     --enable-application-coap         \
-    --enable-border-agent-proxy       \
+    --enable-tmf-tproxy               \
```
Comment (jwhui) [inline examples/Makefile-posix]: typo:  `--enable-tmf-tproxy` -> `--enable-tmf-proxy`

File: include/openthread/Makefile.am [hunk 4]
```diff
@@ -49,7 +49,7 @@ PRETTY_SUBDIRS                          = \
 openthread_headers                      = \
     child_supervision.h                   \
     cli.h                                 \
-    border_agent_proxy.h                  \
+    tmf_proxy.h                           \
```
Comment (jwhui) [inline include/openthread/Makefile.am]: make alphabetical?

**Pull request merged**